File: app/src/main/java/com/example/wpossbank/adaptadores/TransactionLogAdapter.java
package com.example.wpossbank.adaptadores;

import android.content.Context;
import android.os.Build;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.recyclerview.widget.RecyclerView;

import com.example.wpossbank.R;

import java.util.ArrayList;
import java.util.Scanner;

public class TransactionLogAdapter extends RecyclerView.Adapter<TransactionLogAdapter.ViewHolder> {
    private final Context context;
    private final ArrayList<String> entryDate, entryType,entryValue, entrySource;

    public TransactionLogAdapter(Context context, ArrayList<String> entryDate, ArrayList<String> entryType,
                                 ArrayList<String> entryValue, ArrayList<String> entrySource) {
        this.context = context;
        this.entryDate = entryDate;
        this.entryType = entryType;
        this.entryValue = entryValue;
        this.entrySource = entrySource;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int ViewType){
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.layout_log_entry, parent, false);
        return new ViewHolder(view);
    }

    @RequiresApi(api = Build.VERSION_CODES.M)
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        //Introduce el texto correspondiente de cada array en los elementos indicados
        holder.entryDateTxt.setText(String.valueOf(entryDate.get(position)));
        holder.entryTypeTxt.setText(String.valueOf(entryType.get(position)));
        holder.entryValueTxt.setText(String.valueOf(entryValue.get(position)));
        holder.entrySourceTxt.setText(String.valueOf(entrySource.get(position)));

        String valueString = String.valueOf(entryValue.get(position)).replaceAll("[^0-9-]*", "");
        Log.d("TAG", valueString);

        if (Integer.parseInt(valueString) >= 0){
            holder.entryValueTxt.setTextColor(context.getColor(R.color.value_add_money));
        }else{
            holder.entryValueTxt.setTextColor(context.getColor(R.color.value_remove_money));
        }
    }

    @Override
    public int getItemCount() {
        return entryDate.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        //Clase para cada entrada en el recyclerView
        TextView entryDateTxt, entryTypeTxt, entrySourceTxt, entryValueTxt;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            entryDateTxt = itemView.findViewById(R.id.entryDateTxt);
            entryTypeTxt = itemView.findViewById(R.id.entryTypeTxt);
            entryValueTxt = itemView.findViewById(R.id.entryValueTxt);
            entrySourceTxt = itemView.findViewById(R.id.entrySourceTxt);
        }

    }

}

File: app/src/main/java/com/example/wpossbank/adapters/TransactionLogAdapter.kt
package com.example.wpossbank.adapters

import com.example.wpossbank.R
import android.view.LayoutInflater
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import androidx.annotation.RequiresApi
import android.os.Build
import android.content.Context
import android.util.Log
import android.view.View
import androidx.core.content.ContentProviderCompat.requireContext

class TransactionLogAdapter(
    private val context: Context,
    private val entryDate: List<String>,
    private val entryType: List<String>,
    private val entryValue: List<String>,
    private val entrySource: List<String>
) : RecyclerView.Adapter<TransactionLogAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, ViewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.layout_log_entry, parent, false)
        return ViewHolder(view)
    }

    @RequiresApi(api = Build.VERSION_CODES.M)
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        //Introduce el texto correspondiente de cada array en los elementos indicados
        holder.entryDateTxt.text = entryDate[position]
        holder.entryTypeTxt.text = entryType[position]
        holder.entryValueTxt.text = entryValue[position]
        holder.entrySourceTxt.text = entrySource[position]
        val valueString = entryValue[position].replace("[^0-9-]*".toRegex(), "")
        Log.d("TAG", valueString)
        if (valueString.toInt() >= 0) {
            holder.entryValueTxt.setTextColor(context.resources.getColor(R.color.value_add_money))
        } else {
            holder.entryValueTxt.setTextColor(context.resources.getColor(R.color.value_remove_money))
        }
    }

    override fun getItemCount(): Int {
        return entryDate.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        //Clase para cada entrada en el recyclerView
        var entryDateTxt: TextView
        var entryTypeTxt: TextView
        var entrySourceTxt: TextView
        var entryValueTxt: TextView

        init {
            entryDateTxt = itemView.findViewById(R.id.entryDateTxt)
            entryTypeTxt = itemView.findViewById(R.id.entryTypeTxt)
            entryValueTxt = itemView.findViewById(R.id.entryValueTxt)
            entrySourceTxt = itemView.findViewById(R.id.entrySourceTxt)
        }
    }
}